arch/arm/core/aarch32/cortex_m/vector_table.S

```diff
@@ -70,7 +70,11 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
     .word 0
+#if defined(CONFIG_MULTITHREADING)
     .word z_arm_pendsv
```

Comment:
@nordic-krch there is one note here: is it guaranteed that nothing outside #ifdef CONFIG_MULTITHREADING, currently PENDS the PendSV interrupt? Can you check?

Reply:
i've only found `SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;` bwing called by `z_impl_k_thread_abort` and `arch_swap`

Reply:
Are these only compiled-in for MULTITHREADING=y case?

Reply:
I think the PendSV is also set in exception exit for Cortex-M, @nordic-krch can you check that it is not compiled in for !MULTITHREADING?

Reply:
here? https://github.com/zephyrproject-rtos/zephyr/blob/bce0d52664089dd0b2c7dd14e5e7cffd82c7898d/arch/arm/core/aarch32/cortex_m/exc_exit.S#L81
It is only when `CONFIG_PREEMPT_ENABLED` is enabled and it's not when !MULTITHREADING

Reply:
That's what I thought - I wanted you to confirm, as well :)